folium/features.py

```diff
@@ -787,6 +787,134 @@ def __init__(self, location, radius=500, color='black',
             """)
 
 
+class CustomLegend(Layer):
+    """
+    Creates a customizable Legend in one of the corners of the map. Customization highly recommended.
+    this._div.innerHTML can be easily altered (must contain html as a string) but anything can be displayed here with any css style.
+    This class is essential for the use of CustomCircleMarker.
+
+    Parameters
+    ----------
+    content: Content of the Legend box
+    position: position on the screen, default topright
+        e.g. bottomright, topleft, bottomleft
+    style: define a css style for the box, default as given.
+        Input: any css formated string
+    """
+
+    def __init__(self, content, position="topright", style="padding: 6px 8px;font: 14px/16px Arial, Helvetica, sans-serif;box-shadow: 0 0 15px;background: white;border-radius: 5px"):
+        super(CustomLegend,self).__init__()
+        self._name = 'Legend'
+        self.position = position
+        self.content = content
+        self.style = style
+
+        self._template = Template(u"""
+           {% macro script(this, kwargs) %}
+
+            var info = L.control({position: '{{ this.position }}'});
+
+            info.onAdd = function () {
+            this._div = L.DomUtil.create('div', 'info'); // create a div with a class "info"
+            this.update();
+            return this._div;
+            };
+
+            info.update = function (boxtxt) {
+            this._div.innerHTML = (boxtxt ? '<div style="{{ this.style }}">{{ this.content }}' + boxtext + </div> : '<div style="{{ this.style }}">{{ this.content }}</div>')
+            };
+
+            info.addTo({{this._parent.get_name()}});
+
+
+
+        {% endmacro %}
+        """)
+
+
+class CustomCircleMarker(Marker):
```

Comment:
@birdage this is related to your work in #524, right? Can you take a look at this part and join forces with @olibchr.